qa/tasks/fs.py

```diff
@@ -59,21 +60,31 @@ def post_upgrade_checks(ctx, config):
         epoch = mdsmap['epoch']
         pre_upgrade_epoch = fs_state['epoch']
         assert pre_upgrade_epoch < epoch
-        should_decrease_max_mds = fs_state['max_mds'] > 1
+        multiple_max_mds = fs_state['max_mds'] > 1
         did_decrease_max_mds = False
         should_disable_allow_standby_replay = fs_state['flags'] & CEPH_MDSMAP_ALLOW_STANDBY_REPLAY
         did_disable_allow_standby_replay = False
+        did_fail_fs = False
         for i in range(pre_upgrade_epoch+1, mdsmap['epoch']):
             old_status = mdsc.status(epoch=i)
             old_fs = old_status.get_fsmap(fscid)
             old_mdsmap = old_fs['mdsmap']
-            if should_decrease_max_mds and old_mdsmap['max_mds'] == 1:
+            log.debug(f"epoch {i} : MDS not_joinable"
+                      f" = {old_mdsmap['flags'] & CEPH_MDSMAP_NOT_JOINABLE} "
+                      f"and max_mds = {old_mdsmap['max_mds']}")
+            if multiple_max_mds and \
+                    (old_mdsmap['flags'] & CEPH_MDSMAP_NOT_JOINABLE) and\
+                    old_mdsmap['max_mds'] > 1:
+                log.debug(f"max_mds not reduced in epoch {i} as fs was failed "
+                          "for carrying out rapid multi-rank mds upgrade")
+                did_fail_fs = True
```

Comment:
This should only be a valid upgrade path if the mgr config is set. This `fs.py` task should check if that config was set. If the mgr takes this upgrade path without the config set, abort.

Reply:
I can see config in `def post_upgrade_checks(ctx, config)` which it seems I can use, is something like `is_fail_fs = config.get("mgr/orchestrator/fail_fs", False)` possible?